compiler/noirc_frontend/src/parser/errors.rs

```diff
@@ -26,6 +26,8 @@ pub enum ParserErrorReason {
     EarlyReturn,
     #[error("Patterns aren't allowed in a trait's function declarations")]
     PatternInTraitFunctionParameter,
+    #[error("Modifiers are ignored in a trait's function declaration")]
```

Comment:
```suggestion
    #[error("Modifiers are ignored on a trait impl method")]
```
This is a trait impl, not on the trait itself (although those should also be ignored). Trait impl methods are also not declarations, they'd be definitions since they have a body.